src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -321,56 +318,80 @@ private void realExecute(String checkstyleVersion) {
+        handleReturnStatus(warnings, errors);
+    }
 
-        // Handle the return status
-        if (!okStatus) {
-            final String failureMsg =
-                    "Got " + numErrs + " errors (max allowed: " + maxErrors + ") and "
-                            + numWarnings + " warnings.";
+    /**
+     * Processes the result of Checkstyle validation and handles any violations.
+     * If the number of errors or warnings exceeds the configured maximum limits,
+     /**
+     * Processes Checkstyle validation results and handles violations based on configured thresholds.
+     *
+     * <p>Compares the number of errors and warnings against the configured maximum limits.
+     * If either exceeds their respective maximum:
+     * <ul>
+     *   <li>Sets the failure property with an error message (if failureProperty is configured)</li>
+     *   <li>Throws a BuildException (if failOnViolation is true)</li>
+     * </ul>
+     *
+     * @param warnings Counter tracking the number of validation warnings (must not be null)
+     * @param errors Number of validation errors found (must be >= 0)
+     * @throws BuildException if violation thresholds are exceeded and failOnViolation is true
+     * @throws NullPointerException if warnings parameter is null
+     * @throws IllegalArgumentException if errors is negative
+     */
+    private void handleReturnStatus(SeverityLevelCounter warnings, int errors) {
+        if (!(errors <= maxErrors && warnings.getCount() <= maxWarnings)) {
+            final String error = "Got " + errors + " errors (max allowed: " + maxErrors + ") and "
+                    + warnings.getCount() + " warnings.";
             if (failureProperty != null) {
-                getProject().setProperty(failureProperty, failureMsg);
+                getProject().setProperty(failureProperty, error);
             }
-
             if (failOnViolation) {
-                throw new BuildException(failureMsg, getLocation());
+                throw new BuildException(error, getLocation());
             }
         }
     }
 
+    /**
+     * Logs startup information for the Checkstyle process including file location time,
+     * configuration being used, and the number of files to be checked.
+     *
+     * @param checkstyleVersion the version of Checkstyle being used
+     * @return List of files that will be checked
+     */
+    private List<File> logStartup(String checkstyleVersion) {
+        final long start = System.currentTimeMillis();
+        final List<File> files = getFilesToCheck();
+        final long end = System.currentTimeMillis();
+        log("To locate the files took " + (end - start) + TIME_SUFFIX, Project.MSG_VERBOSE);
+        log("Using configuration " + config, Project.MSG_VERBOSE);
+        log("Running Checkstyle "
+                + checkstyleVersion
+                + " on " + files.size()
+                + " files", Project.MSG_INFO);
+        return files;
```

Comment:
grouping the `MSG_VERBOSE` stuff.

Story telling is like we have to gather the config before running of course so why not printing it that way? `Running Checkstyle` seems like the real deal and therefore should give dedicated concern/location.